nomad/structs/workload_id.go

```diff
@@ -203,6 +219,38 @@ func (b *IdentityClaimsBuilder) Build(now time.Time) *IdentityClaims {
 	return claims
 }
 
+func strAttrGet[T any](x *T, fn func(x *T) string) string {
+	if x != nil {
+		return fn(x)
+	}
+	return ""
+}
+
+func (b *IdentityClaimsBuilder) interpolate() {
+
```

Comment:
nbd but might as well avoid creating replacers when there's nothing to replace

```suggestion
if len(b.extras) == 0 {
  return
}
```